src/scss/variables/spacing.scss

```diff
@@ -8,3 +8,110 @@
     --kyn-spacing-gutter: 32px;
   }
 }
+
+//Spacer Definitions
+$spacer-unit-2: 2px;
+$spacer-unit-4: 4px;
+$spacer-unit-8: 8px;
+$spacer-unit-12: 12px;
+$spacer-unit-24: 24px;
+$spacer-unit-32: 32px;
+$spacer-unit-40: 40px;
+$spacer-unit-48: 48px;
+$spacer-unit-64: 64px;
+$spacer-unit-80: 80px;
+$spacer-unit-96: 96px;
+$spacer-unit-128: 128px;
+$spacer-unit-160: 160px;
+
+@mixin spacer($property, $value) {
+  @each $dir in (top, right, bottom, left) {
+    #{$property}-#{$dir}: $value;
+  }
+ }
+ 
+ @mixin spacer-vertical($property, $value) {
+  @each $dir in (top, bottom) {
+    #{$property}-#{$dir}: $value;
+  }
+ }
+ 
+ @mixin spacer-horizontal($property, $value) {
+  @each $dir in (left, right) {
+    #{$property}-#{$dir}: $value;
+  }
+ }
+
+:root {
+// define CSS variables for the spacing values
+ --spacing-0: 0;
+ --spacing-2: #{$spacer-unit-2};
+ --spacing-4: #{$spacer-unit-4};
+ --spacing-8: #{$spacer-unit-8};
+ --spacing-12: #{$spacer-unit-12};
+ --spacing-24: #{$spacer-unit-24};
```

Comment:
1. Prefix all tokens with `--kd-`.
2. I think a token for 16 is very much needed.